03_polars.py

```
@@ -362,40 +344,55 @@ def conditional_join_polars():
+        # Polars simple group by on multiple columns
+        result = order_items.group_by(["order_id", "product_id"]).agg(
+            pl.col("quantity").sum(),
+            pl.col("unit_price").mean(),
+            pl.col("discount_percentage").max(),
         )
         return result
 
     results.append(time_operation("multilevel_groupby", pl, multilevel_groupby_polars))
 
     # Time series resampling
     def time_series_resample_polars():
```

Ruff (0.14.11)

Missing return type annotation for private function `time_series_resample_polars`

(ANN202)